synapse/rest/client/v2_alpha/register.py

```diff
@@ -416,8 +416,11 @@ def on_POST(self, request):
             )
             # Necessary due to auth checks prior to the threepid being
             # written to the db
-            if is_threepid_reserved(self.hs.config, threepid):
-                yield self.store.upsert_monthly_active_user(registered_user_id)
+            if threepid:
```

Comment:
for future ref: I'd be inclined to merge these into one `if` statement, to reduce indentation.